test/unit/parachain/loans.test.ts

```diff
@@ -161,4 +134,181 @@ describe("DefaultLoansAPI", () => {
+
+            const governanceCurrencyBorrowedBtc = governanceCurrencyBorrowedAmount
+                .toBig()
+                .div(exchangeRateGovernanceCurrency);
+            const relayCurrencyBorrowedBtc = relayCurrencyBorrowedAmount.toBig().div(exchangeRateRelayCurrency);
+            const totalBorrowedAmountBtc = governanceCurrencyBorrowedBtc.add(relayCurrencyBorrowedBtc);
+
+            const totalCollateralThresholdAdjustedCollateralAmountBtc = governanceCurrencyCollateralBtc
+                .mul(collateralThresholdGovernanceCurrency)
+                .add(wrappedCurrencyCollateralBtc.mul(collateralThresholdWrappedCurrency));
+            const totalLiquidationThresholdAdjustedCollateralAmountBtc = governanceCurrencyCollateralBtc
+                .mul(liquidationThresholdGovernanceCurrency)
+                .add(wrappedCurrencyCollateralBtc.mul(liquidationThresholdWrappedCurrency));
+
+            const lendPositions = [
+                mockLendPosition(governanceCurrencyLentAmount),
+                mockLendPosition(relayCurrencyLentAmount, false),
+                mockLendPosition(wrappedCurrencyLentAmount),
+            ];
+            const borrowPositions = [
+                mockBorrowPosition(relayCurrencyBorrowedAmount),
+                mockBorrowPosition(governanceCurrencyBorrowedAmount),
+            ];
+
+            const { ltv, collateralThresholdWeightedAverage, liquidationThresholdWeightedAverage } =
+                loansApi.getLendingStats(lendPositions, borrowPositions, loanAssets);
+
+            const expectedLtv = totalBorrowedAmountBtc.div(totalCollateralAmountBtc);
+            const expectedAverageCollateralThreshold =
+                totalCollateralThresholdAdjustedCollateralAmountBtc.div(totalCollateralAmountBtc);
+            const expectedAverageLiquidationThreshold =
+                totalLiquidationThresholdAdjustedCollateralAmountBtc.div(totalCollateralAmountBtc);
+
+            expect(ltv.eq(expectedLtv), `LTV: ${ltv.toString()} does not match expected LTV: ${expectedLtv.toString()}`)
+                .to.be.true;
+            expect(
+                collateralThresholdWeightedAverage.eq(expectedAverageCollateralThreshold),
+                `Average collateral threshold: ${collateralThresholdWeightedAverage.toString()} does not match expected threshold: ${expectedAverageCollateralThreshold.toString()}`
+            ).to.be.true;
+            expect(
+                liquidationThresholdWeightedAverage.eq(expectedAverageLiquidationThreshold),
+                `Average liquidation threshold: ${liquidationThresholdWeightedAverage.toString()} does not match expected threshold: ${expectedAverageLiquidationThreshold.toString()}`
+            ).to.be.true;
+        });
+
+        it("should not throw when there are no positions", () => {
+            expect(() => loansApi.getLendingStats([], [], loanAssets)).to.not.throw;
```

Comment:
Not sure about this, but do we need `await`s here to avoid promises being still evaluated after this test suite is marked as complete?

Reply:
No, in this case there is no need to await, because `getLendingStats` does not return promise.